rust/src/mqtt/mqtt.rs

```diff
@@ -0,0 +1,846 @@
+                    let mut tx = self.new_tx(msg, toclient);
+                    &mut MQTTState::set_event(&mut tx, MQTTEvent::MissingUnsubscribe);
+                    self.transactions.push(tx);
+                }
+            },
+            MQTTOperation::UNASSIGNED => {
+                let mut tx = self.new_tx(msg, toclient);
+                tx.complete = true;
+                &mut MQTTState::set_event(&mut tx, MQTTEvent::UnassignedMsgtype);
+                self.transactions.push(tx);
+            },
+            MQTTOperation::TRUNCATED(_) => {
+                let mut tx = self.new_tx(msg, toclient);
+                tx.complete = true;
+                self.transactions.push(tx);
+            },
+            MQTTOperation::AUTH(_)
+            | MQTTOperation::DISCONNECT(_) => {
+                if !self.connected {
+                    let mut tx = self.new_tx(msg, toclient);
+                    &mut MQTTState::set_event(&mut tx, MQTTEvent::UnintroducedMessage);
+                    self.transactions.push(tx);
+                    return;
+                }
+                let mut tx = self.new_tx(msg, toclient);
+                tx.complete = true;
+                self.transactions.push(tx);
+            },
+            MQTTOperation::PINGREQ
+            | MQTTOperation::PINGRESP => {
+                if !self.connected {
+                    let mut tx = self.new_tx(msg, toclient);
+                    &mut MQTTState::set_event(&mut tx, MQTTEvent::UnintroducedMessage);
+                    self.transactions.push(tx);
+                    return;
+                }
+                let mut tx = self.new_tx(msg, toclient);
+                tx.complete = true;
+                self.transactions.push(tx);
+            }
+        }
+    }
+
+    fn parse_request(&mut self, input: &[u8]) -> AppLayerResult {
+        let mut current = input;
+        if input.len() == 0 {
+            return AppLayerResult::ok();
+        }
+
+        let mut consumed = 0;
+        SCLogDebug!("skip_request {} input len {}", self.skip_request, input.len());
+        if self.skip_request > 0 {
+            if input.len() <= self.skip_request {
+                SCLogDebug!("reducing skip_request by {}", input.len());
+                self.skip_request -= input.len();
+                return AppLayerResult::ok();
+            } else {
+                current = &input[self.skip_request..];
+                SCLogDebug!("skip end reached, skipping {} :{:?}", self.skip_request, current);
+                consumed = self.skip_request;
+                self.skip_request = 0;
+            }
+        }
+
+
+        while current.len() > 0 {
+            let mut skipped = false;
+            SCLogDebug!("request: handling {}", current.len());
+            match parse_message(current, self.protocol_version, self.max_msg_len) {
+                Ok((rem, msg)) => {
+                    SCLogDebug!("request msg {:?}", msg);
+                    match msg.op {
```

Comment:
style : `match` with one case can be an `if`

Reply:
Yep, changed -- thanks.